### PR TITLE
Refactor/reporting requirement schema

### PR DESCRIPTION
**Pull request by BCerki**
[https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1362 and ](https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1362)

and

https://app.zenhub.com/workspaces/climate-action-secretariat-60ca4121764d710011481ca2/issues/gh/bcgov/cas-cif/1363

(more efficient to do them together since they use the same schema)

### CODE REVIEW

File: schema/revert/data/002_insert_json_schema_form_data_remove_default_from_annual_report_schema.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-- Revert cif:data/002_insert_json_schema_form_data_remove_default_from_annual_report_schema from pg
+
+begin;
+
+/* No revert required.
```
Comment (BCerki) [inline schema/revert/data/002_insert_json_schema_form_data_remove_default_from_annual_report_schema.sql:5]: Should there be a revert for these?

File: app/data/jsonSchemaForm/validationSchemas.ts [hunk 2]
```diff
@@ -1,10 +1,8 @@
 import projectContactSchema from "./projectContactSchema";
-import { projectReportingRequirementSchema } from "./projectReportingRequirementSchema";
 import projectManagerSchema from "./projectManagerSchema";
 import projectSchema from "./projectSchema";
 import contactSchema from "./contactSchema";
 import operatorSchema from "./operatorSchema";
-import { milestoneSchema } from "./projectMilestoneSchema";
```
Comment (BCerki) [inline app/data/jsonSchemaForm/validationSchemas.ts:7]: I think this was leftover from the milestone refactor

File: app/components/Form/ProjectQuarterlyReportFormSummary.tsx [hunk 3]
```diff
@@ -50,6 +47,9 @@ const ProjectQuarterlyReportFormSummary: React.FC<Props> = ({
                 formChangeByPreviousFormChangeId {
                   newFormData
                 }
+                formByJsonSchemaName {
```
Comment (tmastrom) [inline app/components/Form/ProjectQuarterlyReportFormSummary.tsx:50]: Is there any difference getting the json schema this way vs `formBySlug`?

Comment (BCerki) [reply app/components/Form/ProjectQuarterlyReportFormSummary.tsx:50]: I don't think so, there are often different ways to get the same piece of info (but if you have concerns, let me know, I'm still new to graphql)

**Review by tmastrom — DISMISSED** [review]
LGTM

**Review by tmastrom — APPROVED** [review]

**Pull request merged**